src/communication_managers/iothub_communication_manager/inc/aduc/iothub_communication_manager.h

```diff
@@ -13,6 +13,40 @@
 
 EXTERN_C_BEGIN
 
+/**
+ * @brief Scans the connection string and returns the connection type related to the string
+ * @details The connection string must use the valid, correct format for the DeviceId and/or the ModuleId
+ * e.g.
+ * "DeviceId=some-device-id;ModuleId=some-module-id;"
+ * If the connection string contains the DeviceId it is an ADUC_ConnType_Device
+ * If the connection string contains the DeviceId AND the ModuleId it is an ADUC_ConnType_Module
+ * @param connectionString the connection string to scan
+ * @returns the connection type for @p connectionString
+ */
+ADUC_ConnType GetConnTypeFromConnectionString(const char* connectionString);
```

Comment:
i'm not 100% sure what belongs in "iothub_communication_manager", but none of these methods seem like they're related to iothub?

Reply:
agreed. Factoring out into a connectioninfo_utils static lib under src/utils seems reasonable to me.

Reply:
@shiyi-peng  please create follow up bug / task for proper refactoring + UT

Reply:
@JeffMill , np , [Task 43982458](https://microsoft.visualstudio.com/OS/_workitems/edit/43982458): [P2] Refactoring iothub_communication_manager functions to a utils file and add UTs thank you!